lib/client.js

```diff
@@ -78,6 +81,11 @@ module.exports = function (dependencies) {
               this.config.token.regenerate(tokenGeneration);
               resolve(this.write(notification, device, retryCount + 1));
               return;
+            } else if (status === "500" && response.reason === "InternalServerError") {
+              stream.connection.close();
+              let error = new VError("Error 500, stream ended unexpectedly");
```

Comment:
@zaumreit That's correct. I'm no maintainer here either, just jumping in to try and lend a hand like yourself. :)

@florianreinhart Happy with making [these changes](https://github.com/node-apn/node-apn/pull/480#discussion_r95142111) or d'ya want to have another PR to deal with that if it's desired?